blockauth/apple/tests/test_web_views.py

```
@@ -222,3 +222,72 @@ def test_callback_clears_cookies_on_error(apple_settings, client):
+    assert response.data["user"]["id"] == "user-1"
+    assert response.data["user"]["email"] == "apple-user@example.com"
+
+
+def test_web_callback_view_subclass_can_override_build_success_response():
+    """A subclass override is reached at the final response builder call site."""
+    from rest_framework.response import Response
+    from blockauth.apple.views import AppleWebCallbackView
+    from blockauth.utils.social import SocialLoginResult
+
+    class _SwapResponse(AppleWebCallbackView):
+        def build_success_response(self, request, result):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `build_success_response`

(ANN202)